tools/statplot.py

```diff
@@ -1,229 +1,305 @@
-
-# Fire up the GUI
-gui.DisplayWindow(window)
+#!/usr/bin/env python2
+
+import gi
+gi.require_version('Gdk', '3.0')
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gdk, Gtk  # noqa: E402
+
+import argparse  # noqa: E402
+import fluidity.diagnostics.fluiditytools as fluidity_tools  # noqa: E402
+from matplotlib.backends.backend_gtk3agg import (
+    FigureCanvasGTK3Agg as FigureCanvas)  # noqa: E402
+from matplotlib.backends.backend_gtk3 import (
+    NavigationToolbar2GTK3 as NavigationToolbar)  # noqa: E402
+import matplotlib.pyplot as plt  # noqa: E402
+from matplotlib import rc  # noqa: E402
+import matplotlib.ticker as tck  # noqa: E402
+import numpy  # noqa: E402
+import sys  # noqa: E402
+import time  # noqa: E402
+import warnings  # noqa: E402
+
+rc('text', usetex=False)
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description='GUI for Fluidity .stat outputs\n\nWhen the window is '
+    + 'displayed, press the following keys for additional options:\n'
+    + '- x, y -> Change the scale of the x, y axis (automatically chooses '
+    + 'between linear, log and symlog).\n- l -> Change the representation of '
+    + 'the data (solid line or markers).\n- r -> Reload the .stat file. Only '
+    + 'relevant for simulations that are still running.\n- Esc -> Release '
+    + 'focus on text boxes.\n- q -> Exit the GUI.')
+parser.add_argument('statfile', nargs='+', help='path to the .stat file')
+args = parser.parse_args(sys.argv[1:])
+
+
+class StatplotWindow(Gtk.Window):
+    def __init__(self, statfile):
+        Gtk.Window.__init__(self)
+        self.connect('key-press-event', self.KeyPressed)
+        self.set_border_width(8)
+        self.set_default_size(1600, 900)
+        self.set_position(Gtk.WindowPosition.CENTER)
+        self.set_title(statfile[-1])
+        self.statfile = statfile
+        self.vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=8)
+        self.vbox.set_homogeneous(False)
+        self.add(self.vbox)
+        hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=8)
+        hbox.set_homogeneous(True)
+        self.vbox.pack_end(hbox, False, False, 0)
+        self.entries = self.ReadData(statfile)
+        self.xCombo = Gtk.ComboBoxText.new_with_entry()
+        self.xCombo.set_wrap_width(3)
+        self.yCombo = Gtk.ComboBoxText.new_with_entry()
+        self.yCombo.set_wrap_width(3)
+        self.PopulateCombo('load')
+        self.IniCompl(self.xCombo)
+        self.IniCompl(self.yCombo)
+        self.xCombo.connect('changed', self.ComboChangedX)
+        self.xCombo.connect('key-release-event', self.ReleaseFocus)
+        self.yCombo.connect('changed', self.ComboChangedY)
+        self.yCombo.connect('key-release-event', self.ReleaseFocus)
+        hbox.pack_start(self.xCombo, True, True, 0)
+        hbox.pack_start(self.yCombo, True, True, 0)
+        self.fig, self.ax = plt.subplots(nrows=1, ncols=1, num=0,
```

Comment:
This is something weird on one of my systems, but creating the figure this way gives me a segfault somewhere in the depths of gtk. Instead, something like the following works:
```suggestion
        from matplotlib.figure import Figure
        ...
        self.fig = Figure(figsize=(14,7))
        self.ax = self.fig.gca()
```

Reply:
Just out of curiosity, which versions of gi and matplotlib is this happening on?

Reply:
gi 3.32.1, matplotlib 3.1.1

I think using the OO-interface in this context is the right thing to do regardless.